pkg/R/sparkR.R

```diff
@@ -121,45 +119,38 @@ sparkR.init <- function(
   if (sparkRExistingPort != "") {
     sparkRBackendPort <- sparkRExistingPort
   } else {
+    # TODO: test the random port and retry, or use httpuv:::startServer
+    callbackPort = sample(40000, 1) + 10000
     if (Sys.getenv("SPARKR_USE_SPARK_SUBMIT", "") == "") {
       launchBackend(classPath = cp,
                     mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
-                    args = as.character(sparkRBackendPort),
+                    args = as.character(callbackPort),
                     javaOpts = paste("-Xmx", sparkMem, sep = ""))
     } else {
       # TODO: We should deprecate sparkJars and ask users to add it to the
       # command line (using --jars) which is picked up by SparkSubmit
       launchBackendSparkSubmit(
           mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
-          args = as.character(sparkRBackendPort),
+          args = as.character(callbackPort),
           appJar = .sparkREnv$assemblyJarPath,
           sparkHome = sparkHome,
           sparkSubmitOpts = Sys.getenv("SPARKR_SUBMIT_ARGS", ""))
     }
+    sock <- socketConnection(port = callbackPort, server = TRUE, open = 'rb',
```

Comment:
Can we bring up this server socket before we launch the backend ? I just want to avoid the race condition where the backend tries to open a socket and the server socket was not created for some reason.